src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal.test.ts

```diff
@@ -0,0 +1,539 @@
+    expect(vdom.getByAltText('CROMWELL'));
+  });
+  // populated envs
+  it('Renders populated cloud environment', () => {
+    // Arrange
+    const CloneCEM = {
+      ...CloudEnvironmentModalDefaultProps,
+      runtimes: [getGoogleRuntime()],
+      apps: [generateTestApp({})],
+      appDataDisks: [defaultTestDisk],
+    };
+    // Act
+    const vdom = render(h(CloudEnvironmentModal, CloneCEM));
+    // Assert
+    expect(vdom.getByText('Cloud Environment Details'));
+    expect(vdom.getByAltText('Jupyter'));
+    expect(vdom.getByAltText('RStudio'));
+    expect(vdom.getByAltText('GALAXY'));
+    expect(vdom.getAllByText(/Running/).length).toBe(2);
+    expect(vdom.getAllByText('Pause').length).toBe(3);
+    expect(vdom.getAllByText('Open').length).toBe(3);
+    // expect(vdom.getAllByText('Edit existing Environment').length).toBe(2)
+    // expect(vdom.getAllByText('Edit existing Environment')[0]).not.toBeVisible()
+  });
+
+  it('Renders populated azure cloud environment modal', () => {
+    // Arrange
+    const cloneAzure = {
+      ...AzureCloudEnvironmentModalDefaultProps,
+      workspace: defaultAzureWorkspace,
+      persistentDisks: [azureDisk],
+    };
+    // Act
+    const vdom = render(h(CloudEnvironmentModal, cloneAzure));
+    // Assert
+    expect(vdom.getByText('Cloud Environment Details'));
+    expect(vdom.getByAltText('JupyterLab'));
+    expect(vdom.getByAltText('CROMWELL'));
+    expect(vdom.getAllByText('Pause').length).toBe(1);
+    expect(vdom.getAllByText('Open').length).toBe(2);
+    expect(vdom.getAllByText('No Environment found').length).toBe(2);
+    expect(vdom.getAllByText('No Environment found')[0]).not.toBeVisible();
+  });
+
+  // button tests(pause)
+  it.each([
+    {
+      // Jupyter
+      input: {
+        ...CloudEnvironmentModalDefaultProps,
+        runtimes: [getGoogleRuntime()],
+        apps: [
+          generateTestApp({
+            appName: 'app1',
+            status: 'RUNNING',
+          }),
+        ],
+        appDataDisks: [defaultTestDisk],
+      },
+      expectedOutput: {
+        buttonIndex: 0,
+        stopTimes: 1,
+        pauseTimes: 0,
+      },
+    },
+    {
+      // RStudio
+      input: {
+        ...CloudEnvironmentModalDefaultProps,
+        runtimes: [
+          getGoogleRuntime({
+            tool: tools.RStudio,
+          }),
+        ],
+        appDataDisks: [defaultTestDisk],
+      },
+      expectedOutput: {
+        buttonIndex: 1,
+        stopTimes: 1,
+        pauseTimes: 0,
+      },
+    },
+    {
+      // Galaxy
+      input: {
+        ...CloudEnvironmentModalDefaultProps,
+        runtimes: [getGoogleRuntime()],
+        apps: [galaxyRunning],
+        appDataDisks: [defaultTestDisk],
+      },
+      expectedOutput: {
+        buttonIndex: 2,
+        stopTimes: 0,
+        pauseTimes: 1,
+      },
+    },
+  ])(
+    'Invokes ajax call for pause button on a populated google cloud environments instance',
```

Comment:
I would include some information from the test case in the test string.
So I would fit in Jupyter | Galaxy | RStudio somewhere in the title, just so they're listed as different tests in the test output.